polling_stations/apps/addressbase/management/commands/create_dummy_cleaned_addresses_from_onspd.py

```diff
@@ -0,0 +1,97 @@
+import csv
+from pathlib import Path
+
+from django.core.management.base import BaseCommand
+from django.db.models import F, Value, CharField, OuterRef, Subquery, Func
+from django.db.models.functions import Concat, Replace
+
+from addressbase.models import Address
+from uk_geo_utils.models import Onspd
+
+
+class AsEWKT(Func):
+    """Calls the postgis ST_AsEWKT function to get strings like: SRID=4326;POINT(0.088849 51.43053)"""
+
+    function = "ST_AsEWKT"
+    output_field = CharField()
+
+
+class Command(BaseCommand):
+    help = (
+        "Export postcodes that exist in ONSPD but not in AddressBase to a CSV file."
+        "It's a good idea to name the output file according to the version of onspd it's derived from."
+        "eg: python manage.py create_dummy_cleaned_addresses_from_onspd ONSPD_AUG_2024_addressbase_cleaned.csv"
+    )
+
+    def add_arguments(self, parser):
+        parser.add_argument("output_file", type=str, help="Path to output CSV file")
+
+    def handle(self, *args, **options):
+        output_file = options["output_file"]
+
+        output_path = Path(output_file)
+        output_path.parent.mkdir(parents=True, exist_ok=True)
+
+        # First, create a queryset for postcodes that exist in the Address model
+        existing_postcodes = Address.objects.filter(postcode=OuterRef("pcds")).values(
+            "postcode"
+        )
+
+        # Main query
+        results = (
+            Onspd.objects.annotate(
+                # Create "ONSPD:{postcode}" with whitespace removed
+                uprn=Concat(
+                    Value("ONSPD-", output_field=CharField()),
+                    Replace("pcds", Value(" "), Value(""), output_field=CharField()),
+                    output_field=CharField(),
+                ),
+                address=Value("", output_field=CharField()),
+                postcode=F("pcds"),
+                location_wkt=AsEWKT("location"),
+                address_type=Value("D", output_field=CharField()),
```

Comment:
Do these actually correspond to Type D addresses?
My instinct is it would be better to use something like `Z` here that definitely isn't a valid value in the same way that we decided to make "uprn" obviously not a UPRN

Reply:
Yeah, I guess it probably should be something different. I'm not sure it matters in practice, because the value is only there to make it correspond with the glue table schema. I'm sort of tempted to make it blank, similar to the address value. Or maybe set both to `N/A`?